resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -259,12 +259,19 @@ private[spark] object KubernetesUtils extends Logging {
       isLocalDependency(Utils.resolveURI(resource))
   }
 
-  def renameMainAppResource(resource: String, conf: SparkConf): String = {
+  def renameMainAppResource(
+      resource: String,
+      conf: Option[SparkConf] = None,
+      isJava: Boolean = true): String = {
```

Comment:
Can we use `language-neural` parameter name, @skonto ? Then, this function can be reused for `SparkR` later.

Reply:
Also, let's not have the default value. This function is used only twice. And you changed both places in this PR. There is no reason to introduce a new default value.

Reply:
@dongjoon-hyun ok sure.